src/charm.py

```diff
@@ -288,40 +170,26 @@ def _on_tracing_broken(self, _):
         """Update tracing relations' databags once one relation is removed."""
         self._update_tracing_relations()
 
-    def _on_cert_handler_changed(self, _):
-        if self.tls_available:
-            logger.debug("enabling TLS")
-            self.nginx.configure_tls(
-                server_cert=self.cert_handler.server_cert,  # type: ignore
-                ca_cert=self.cert_handler.ca_cert,  # type: ignore
-                private_key=self.cert_handler.private_key,  # type: ignore
-            )
-        else:
-            logger.debug("disabling TLS")
-            self.nginx.delete_certificates()
+    def _on_cert_handler_changed(self, e: ops.RelationChangedEvent):
+
+        self.coordinator._on_cert_handler_changed(e)
```

Comment:
I would expect the coordinator to do that on its own while handling the event? 
This handler will be called after the coordinator's one runs, so we should be covered.